Mydev

oldzhu opened the pull request:
Hi Adrinan,

Committed the changes and ask for review. Let me know if there is any problem or question.

This time, I added one below file

aamt.pl
modified 3 below files:
aamt_auth.pm
aamt_parse.pm
aamt_parse2.pm
deleted 2 below files:
aamt_deploysite.pl
aamt.listsites.pl

You can simply run perl aamt.pl to see the usage as the below:

Usage: sudo perl aamt.pl [-i|--interactive]\n"); 
Usage: sudo perl aamt.pl [-l|--list configfile]\n"); 
Usage: sudo perl aamt.pl [-d|--deploy  configfile siteindex publishprofile]\n"); 
Usage: perl aamt.pl [--help|-?]\n"); 
configfile: The apache main config file\n"); 
siteindex: The index of the source web site and should be number\n"); 
publishprofile: The publishing profile file of the target azure web site\n"); 

Hi Adrian,
As we discussed, I made some changes and please review. Let me know if there is any problem or question.

modified file: 
aamt_parse2.pm

new added files:
aamt_deploysite.pl,
aamt_listsites.pl,
framework.def,
readphp.def,
writephp.def,
read_dzuc_settings.php
write_wp.php
write_joomla.php
write_drupal.php
write_dzuc.php

oldzhu committed:
First commnit to AAMT

oldzhu committed:
First commnit to AAMT

theadriangreen commented on aamt_deploysite.pl:
```diff
@@ -0,0 +1,159 @@
+};
+if($@)
+{
+    #if ($@ !~ /EXIT_TOOL_NOW/)
```
Why is this commented out?

theadriangreen commented on aamt_deploysite.pl:
```diff
@@ -0,0 +1,159 @@
+sub DeleteWorkingFolder
+{
+    my $strYesOrNo = "";
+#    while($strYesOrNo!~/^\s*[YynN]\s*$/)
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip retries, if it is false, then keep the current logic of asking the user for input.

theadriangreen commented on aamt_listsites.pl:
```diff
@@ -0,0 +1,159 @@
+
+    # my @files        = grep { -f } glob( '*.conf' );
+    my @files = File::Find::Rule->file()
+        ->name("*apache*")            
```
This will not work if the config file name is httpd.conf, which it commonly is.

theadriangreen commented on aamt_listsites.pl:
```diff
@@ -0,0 +1,159 @@
+sub DeleteWorkingFolder
+{
+    my $strYesOrNo = "";
+#    while($strYesOrNo!~/^\s*[YynN]\s*$/)
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip retries, if it is false, then keep the current logic of asking the user for input.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -157,6 +163,244 @@ sub pars_GetDirlistinght
+                        return 1;
+                }
+            }
+return 0;
```
Fix indentation

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -157,6 +163,244 @@ sub pars_GetDirlistinght
+		$settingsLine=$settingsLine.$line;	
+	}	
+	if($lineNotMatch)
+	{return (qr/$lineMatch/,qr/$lineNotMatch/,$settingsLine);}   
```
Fix the style here

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -157,6 +163,244 @@ sub pars_GetDirlistinght
+	if($lineNotMatch)
+	{return (qr/$lineMatch/,qr/$lineNotMatch/,$settingsLine);}   
+	else
+	{return (qr/$lineMatch/,$lineNotMatch,$settingsLine);}
```
Fix the style here

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -157,6 +163,244 @@ sub pars_GetDirlistinght
+
+sub pars_DisplaySites
+{
+my $i;
```
Fix indentation

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -157,6 +163,244 @@ sub pars_GetDirlistinght
+	ilog_printf(1,"[$framework]:[$strSiteName]:[$documentRoot]");
+	ilog_printf(1,"\n");
+    }
+#    ui_printline();
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip printing.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -230,7 +477,8 @@ sub pars_CreateReadinessReport
             my $dbHost = $array[$i][DB_HOST];
             my $dbPassword = $array[$i][DB_PASS];
-
+	    die "Didn't get MYSQL password...\n" if(!$dbPassword);
```
Add retry if running interactively.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -296,7 +544,7 @@ sub pars_CreateReadinessReport
 
         my $publishSuccess = FALSE;
-        while (!$publishSuccess)
+#        while (!$publishSuccess)
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip retries, if it is false, then keep the current logic of asking the user for input.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -308,12 +556,13 @@ sub pars_CreateReadinessReport
-                while($strYesOrNo!~/^\s*[YynN]\s*$/)
+#                while($strYesOrNo!~/^\s*[YynN]\s*$/)
                 {
-                    ilog_printf(1, "    Would you like to retry uploading the readiness report? (Y/N):");
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip retries, if it is false, then keep the current logic of asking the user for input.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -323,8 +572,9 @@ sub pars_CreateReadinessReport
 
-        ilog_print(1,"\nReadiness report uploaded, to continue navigate to:\n ${SITE_URL}/results/index/$guid\n\nCreate site and databases and then download and save the publish settings file to this computer.");
-        
+#        ilog_print(1,"\nReadiness report uploaded, to continue navigate to:\n ${SITE_URL}/results/index/$guid\n\nCreate site and databases and then download and save the publish settings file to this computer.");
```
If running interactively this should be displayed.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -355,16 +605,17 @@ sub pars_UploadPublishSettingsAllSites
-    ui_printline();
-    ilog_printf(1,"Please enter the location of the publish settings: \n");
-    ui_printline();
+#    ilog_print(1,"\n\n");
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can get the information from the cmd line, else keep the current logic of asking the user for input.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -408,8 +660,7 @@ sub pars_UploadPublishSettingsAllSites
         ilog_setLogInformation('INT_INFO',$recoveryFile,MSG_FILE_CLOSE,'');
     }
-
-    ilog_print(1,"\nThanks for using the Apache to Azure App Service Migration Tool! \n");
```
Why is this changed?

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -482,10 +734,11 @@ sub pars_PublishSite
             $strYesOrNo =" ";
             while($strYesOrNo!~/^\s*[YynN]\s*$/)
             {
-                ilog_printf(1, "    Would you like to retry publishing the site [$strSiteName]? (Y/N):");
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can get the information from the cmd line, else keep the current logic of asking the user for input.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -512,7 +765,12 @@ sub pars_PublishSite
         if ($tempSplit[0] =~ /pwd/i || $tempSplit[0] =~ /password/i)
         {
-            $rPassword = $tempSplit[1];
+		my $tempValue = $tempSplit[1];
```
Fix indentation.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -529,15 +787,37 @@ sub pars_PublishSite
-            my $dbHost = $array[$sIndex][DB_HOST];
-            my $dbPass = $array[$sIndex][DB_PASS];
-            my $wpSiteurl = $array[$sIndex][WP_SITEURL];
+if ($rUsername eq " " || !$array[$sIndex][MYSQL])
```
Fix indentation.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -529,15 +787,37 @@ sub pars_PublishSite
+        mkdir "$workingFolder/wwwroot";
+
+	foreach my $cfgFile (@cfgFiles)
+	{           # local settings read from config file
```
Put comment on its own line.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -645,42 +906,46 @@ sub pars_PublishSite
             # DUMP MYSQL
             my $retries = 0;
-            while($returnCode != 0 && $retries < 5)
+            while($returnCode != 0 && $retries < 1)
```
Why is this changed?

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -645,42 +906,46 @@ sub pars_PublishSite
             {
                 ilog_print(1,"\nBacking up database on: $dbHost...\n");
-                if ($DEBUG_MODE) { ilog_print(1,"\nmysqldump --single-transaction -h $dbHost -u $dbUser -p'$dbPass' $dbName > $workingFolder/mysqldump.sql\n"); }
+{ ilog_print(1,"mysqldump --single-transaction -h $dbHost -u $dbUser -p'dbPass' $dbName > workingFolder/mysqldump.sql\n\n"); }
```
1. Fix indentation.
2. Shouldn't this be $workingFolder/mysqldupmp.sql?

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -645,42 +906,46 @@ sub pars_PublishSite
 
                 $returnCode = $?;
                 $retries++;
-                if ($returnCode != 0 && $retries < 5)
```
Why is this changed?

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -645,42 +906,46 @@ sub pars_PublishSite
+		ilog_print(1,"\nplease check and try moving databsae manually...\n");
                 while($strYesOrNo!~/^\s*[YynN]\s*$/)
                 {
-                    ilog_printf(1, "    Would you like to retry publishing the database for site [$strSiteName]? (Y/N):");
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip retries, if it is false, then keep the current logic of asking the user for input.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -873,7 +1136,7 @@ sub deployToSite
         }
         
-        ilog_print(1,"[" . "=" x $nBar . " " x ($barWidth - $nBar) . "]\r");
+#        ilog_print(1,"[" . "=" x $nBar . " " x ($barWidth - $nBar) . "]\r");
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip this, if it is false, then keep the current code.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -883,10 +1146,18 @@ sub deployToSite
     if ($DEBUG_MODE) { ilog_print(1,"\nDEBUG: content upload response code: $rCode\n"); }
     if ($DEBUG_MODE) { ilog_print(1,"\nDEBUG: content upload result: $rContent\n"); }
-    ilog_print(1,"\n$itemToAdd published with response code: $rCode\nTo site: $publishUrl\n");
+    if(index($itemToAdd,'test01/wwwroot') != -1)
```
Can you add a comment about what this check is doing?

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -941,7 +1212,7 @@ sub pars_GetFileFromSource
 {   
     utf_setCurrentModuleName('PARSER');
-    ilog_print(1,"\n\nParsing Apache conf file(s)... ");
+#    ilog_print(1,"\n\nParsing Apache conf file(s)... ");
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip this, if it is false, then keep the current code.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -4713,12 +4960,13 @@ sub pars_siteHasValidFrameworkDb
-        while($strYesOrNo!~/^\s*[YynN]\s*$/)
+#        while($strYesOrNo!~/^\s*[YynN]\s*$/)
         {
-            ilog_printf(1, "\n    $framework site detected, would you like to automatically create and migrate the database [$strSiteName]? (Y/N):");
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip this, if it is false, then keep the current code.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -4908,21 +5114,24 @@ sub pars_AskSelectSites
     }
 
     $array[$sIndex][MYSQL] = 0;
-    my $promptyn;    
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip this, if it is false, then keep the current code.

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -4908,21 +5114,24 @@ sub pars_AskSelectSites
-    while($strYesOrNo!~/^\s*[YynN]\s*$/)
+#    while($strYesOrNo!~/^\s*[YynN]\s*$/)
     {           
-        ilog_printf(MSG_MIGRATE_SITE, $strSiteName);
```
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip this, if it is false, then keep the current code.

theadriangreen commented on aamt_deploysite.pl:
```diff
@@ -0,0 +1,159 @@
+use aamt_parse;             #used for parsing the conf file and getting the site information	
+use aamt_parse2;		    #used for parsing the conf file and generating the 2D array.
+
+# main subroutine starts here
```
Add something to display the command line usage when running in unattended mode. i.e. if you run the tool with "help" or "/?" or man it should show the sample usage with a description.

oldzhu replied:
Plan to merge the aamt_deploysite.pl,aamt_listsites.pl and the original migrate_tool_main.pl into one main script like aamt.pl, it works as the below:
1. Display usage help when run
sudo perl aamt.pl 
or
sudo perl aamt.pl -help
2. Keep the original interactive behavior when run
sudo perl aamt.pl -i
or
perl aamt.pl -I
3. Run in non-interactive when run
sudo perl aamt.pl list test01 /etc/apache2/apache2.conf  
or
sudo perl aamt.pl deploy test01 /etc/apache2/apache2.conf 2 /home/azureadmin/testmig102.chinacloudsites.cn.PublishSettings
Will got ahead to implement after get your comments or suggestions.

theadriangreen replied:
Sounds good!

oldzhu committed:
Modify to support both interactive and non-interactive mode in one main script

theadriangreen commented on aamt.pl:
```diff
@@ -0,0 +1,454 @@
+use strict;
```
aamt.pl should be deleted and the changes should be applied to migrate_tool_main.pl

oldzhu committed:
rename aamt.pl to migrate_tool_main.pl

theadriangreen commented on aamt_auth.pm:
```diff
@@ -38,21 +38,29 @@ my  $REC_MY_NAME_VIRTUAL_HOST ='[NAME_VIRTUAL_HOST]' ;
 {
     ui_clearScreen();         #used to clear screen defined in aamt_userinterface.pm file
-	ui_Title();               #used for title of the migartion kit defined in aamt_userinterface.pm file	
+    ui_Title() if(&utf_getRunMode() eq 'i');             #used for title of the migartion kit defined in aamt_userinterface.pm file
```
use consistent coding style for blocks:
if (&utf_getRunMode() eq 'i')
{
    ui_Title();
}

theadriangreen commented on aamt_parse2.pm:
```diff
@@ -269,7 +518,11 @@ sub pars_CreateReadinessReport
     }
-    
+
+    if(&utf_getRunMode() eq 'd'){
```
consistent coding style as above

theadriangreen commented on migrate_tool_main.pl:
```diff
@@ -36,9 +47,108 @@
+{
+    ilog_print(1,"Unknown option:@_\n") if (@_);
+    ilog_print(1,"Usage: sudo perl $0 [-i|--interactive]\n");
+    ilog_print(1,"Usage: sudo perl $0 [-l|--list configfile]\n");
```
what does this mode do?

theadriangreen commented on migrate_tool_main.pl:
```diff
@@ -36,9 +47,108 @@
+    exit 0;
+}
+
+sub modeh
```
Can you give these subroutines better names? i.e. sub modeHelp. Also, can you make the mode string more than just a single letter? It is hard to read.

theadriangreen commented on migrate_tool_main.pl:
```diff
@@ -36,9 +47,108 @@
+
+sub model
+{
+    if($mode && $mode  ne 'l') {&usage();}
```
'l' is a bad name for a variable since it's hard to understand what it means and it looks like '1' or 'I'

oldzhu committed:
changed the short mode name to long
"h" to "help"
"i" to "interactive"
"l" to "list"
"d" to "deploy"
Changed the code to support "originalsitename" in "interactive" mode.

oldzhu committed:
Having different behavior based on how the tool is invoked is bad and
modify to inject the "originalsitename" into the publish profile using
the perl script if it's not there.
add the two shell script which will be called by windows frontend program

theadriangreen commented on migrate_tool_main.pl:
```diff
@@ -151,15 +435,20 @@ sub TerminateTool
     my $strYesOrNo = "";
-    while($strYesOrNo!~/^\s*[YynN]\s*$/)
-    {
-        ilog_printf(1, "    Would you like to delete the working folder used to store temporary settings? (Y/N):");
```
Don't automatically delete the working folder. Keep the prompt.

oldzhu committed:
fix the two problems
1. The problem to detect DB connection string for Drupal7.
2. The probblem to modify WP_CONTENT_DIR setting in wordpress

oldzhu committed:
Add the project files of the windows frontend program.

oldzhu committed:
remove aamt.pl
